### ISSUE TITLE
Javascript package/library dependency manager

### ISSUE THREAD

**Issue by gkanwar** [learning-unlimited/ESP-Website#760]
Currently our JS external library dependencies are not well-maintained. We should consider doing this. See, for example: http://bower.io/. We could also use http://requirejs.org/ for JS module dependencies.

**Comment by benjaminjkraft** [comment]
This would, among other things, potentially make it a lot easier to know where we're using JS deps to make them easier to upgrade.

**Comment by HackerAaditya** [comment]
Hi @jmoldow @willgearty , I reviewed this issue and analyzed the current JavaScript dependency setup. I propose to modernize it by introducing npm for dependency management, organizing external libraries, and setting up a build pipeline.

Instead of Bower/RequireJS, I suggest using npm along with a modern bundler (like Vite) to improve maintainability and scalability, while migrating gradually to avoid breaking existing functionality.

May I proceed with this approach? If approved, please assign this issue to me.

**Comment by HackerAaditya** [comment]
Yes i want this issue to be assigned it to me .  But i have suggested using npm along with vite . 

/assign

**Comment by willgearty** [comment]
Unassigning due to inactivity.

**Comment by sanketkokate712** [comment]
Hi @willgearty and team!

I was looking through older infrastructure issues and found this one from 2013, where @gkanwar pointed out that our JavaScript libraries are hard to keep track of. @benjaminjkraft later said the same thing: if we knew exactly what we were using and where, upgrades would be a lot less painful.

That still feels true. I looked around a bit: the theme editor already has a package.json with Bootstrap and friends, but a lot of the rest of the site still pulls jQuery and other scripts from CDNs or from copies sitting in /media/scripts. There’s no one list you can glance at when something needs an update.

I wouldn’t try to rebuild the whole frontend in one go (Bower/RequireJS, or a big Vite rewrite). I’d start smaller:
Make a short inventory of the JS we actually load (jQuery, jQuery UI, and the plugins on things like onsite and the editor).
Put those versions in one place, similar to what you already do for the theme editor, so the next person isn’t hunting through templates.
Move a couple of the common libraries over first, and leave pages working the way they do today. No “everything must go through a bundler” step unless you decide that’s worth it later.

Would that kind of first pass still be useful? If so, could you assign this issue to me?
